flytekit/types/pickle/pickle.py

```diff
@@ -99,7 +98,9 @@ def to_literal(self, ctx: FlyteContext, python_val: T, python_type: Type[T], exp
                 format=self.PYTHON_PICKLE_FORMAT, dimensionality=_core_types.BlobType.BlobDimensionality.SINGLE
             )
         )
-        remote_path = FlytePickle.to_pickle(python_val)
+        remote_path = FlytePickle.to_pickle(ctx, python_val)
+        if not ctx.file_access.is_remote(remote_path):
+            remote_path = ctx.file_access.put_raw_data(remote_path)
```

Comment:
We can remove these two lines, right? `FlytePickle.to_pickle` should upload the data to s3.

Reply:
Yes, you are correct. I removed it.